src/inmanta/agent/forking_executor.py

```diff
@@ -144,6 +146,19 @@ def _detach_log_shipper(self) -> None:
             logging.getLogger().removeHandler(self.log_transport)
             self.log_transport = None
 
+    def start_timer_venv_checkup(self, interval: int) -> None:
+        self.timer_venv_checkup = util.Scheduler("venv_checkup_scheduler")
+        self.timer_venv_checkup.add_action(
+            action=self.touch_inmanta_venv_status,
+            schedule=util.IntervalSchedule(
+                interval=interval,
+            ),
+        )
+
+    def stop_timer_venv_checkup(self) -> None:
+        if self.timer_venv_checkup is not None:
+            asyncio.ensure_future(self.timer_venv_checkup.stop())
```

Comment:
I am not sure what to do with this. Like this the `stop()` method will only get called when the ioloop picks up the task. This might never happen. The annoying thing is that the `connection_lost()` method is a sync method, so there is no easy way to make sure that the `self.timer_venv_checkup.stop()` call gets awaited. @sanderr Do you see an easy way out here?